### PR TITLE
9576 vs coil model sets heating capacity

### LINKED ISSUE #9576: The Coil:Cooling:DX:VariableSpeed model sets the heating coil capacity the same regardless if it is a DX coil.
Issue overview
--------------
A heat pump has only 1 coil and, given the model uses 2 seperate coils, the capacity of those coils should match, save differences in compressor heat for heating mode. Of course the output of that coil will change from cooling to heating based on the operating conditions (outdoor air temp) but it is still one coil. The variable speed coil model assumes the companion heating coil is also a variable speed coil and sizes that coil to match the cooling capacity. When a non-DX heating coil is used, that means there really are 2 coils and the size of those coils could be different.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (5Zone_Unitary_VSDesuperheater)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes #9576
 - Created to fix issue with VS coil model and new coil model
 - The VS coil model always sizes the heating coil to match the cooling capacity. For non-HP heating coil, the heating capacity should match the load

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/UnitarySystem.cc [hunk 1]
```diff
@@ -2222,7 +2222,9 @@ namespace UnitarySystems {
             state.dataSize->DXCoolCap = VariableSpeedCoils::GetCoilCapacityVariableSpeed(
                 state, DataHVACGlobals::cAllCoilTypes(this->m_CoolingCoilType_Num), this->m_CoolingCoilName, ErrFound);
             EqSizing.DesCoolingLoad = state.dataSize->DXCoolCap;
-            EqSizing.DesHeatingLoad = state.dataSize->DXCoolCap;
+            if (this->m_DXHeatingCoil) {
+                EqSizing.DesHeatingLoad = state.dataSize->DXCoolCap;
+            }
```
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.cc:2227]: Why is there ever a reason to do in 3 lines what you could do in 1?

**Comment by rraustad** [comment]
Both diffs have a gas heating coil with VS cooling coil. For heat pumps cooling and heating should match. For non-HPs cooling and heating should size to respective capacity to meet load.

      Component Sizing Information, AirLoopHVAC:UnitarySystem, GASHEAT DXAC FURNACE 1, Design Size Nominal Cooling Capacity [W], 53882.19699
    - Component Sizing Information, AirLoopHVAC:UnitarySystem, GASHEAT DXAC FURNACE 1, Design Size Nominal Heating Capacity [W], 53882.19699
    + Component Sizing Information, AirLoopHVAC:UnitarySystem, GASHEAT DXAC FURNACE 1, Design Size Nominal Heating Capacity [W], 28151.00449
      Component Sizing Information, AirLoopHVAC:UnitarySystem, GASHEAT DXAC FURNACE 1, Design Size Maximum Supply Air Temperature [C], 18.00000
      Component Sizing Information, AirLoopHVAC:UnitarySystem, GASHEAT DXAC FURNACE 1, User-Specified Maximum Supply Air Temperature [C], 80.00000
      Component Sizing Information, AirLoopHVAC:UnitarySystem, GASHEAT DXAC FURNACE 1, User-Specified Supplemental Heating Coil Nominal Capacity [W], 0.00000
      Component Sizing Information, Fan:VariableVolume, SUPPLY FAN 1, Design Size Maximum Flow Rate [m3/s], 2.62064
    - Component Sizing Information, Coil:Heating:Fuel, MAIN HEATING COIL 1, Design Size Nominal Capacity [W], 53882.19699
    + Component Sizing Information, Coil:Heating:Fuel, MAIN HEATING COIL 1, Design Size Nominal Capacity [W], 28151.00449

**Comment by rraustad** [comment]
I checked that the smaller heating coil could meet the load in winter. Since the coil is smaller the Unitary System Part Load Ratio is now higher but the Main Heating Coil 1 Heating Coil NaturalGas Rate has not changed, so the heating coil met the same load. This is from 5Zone_Unitary_VSDesuperheater with Chicago Ohare Intl Ap. A critically sized coil will operate near PLR=1 at design load. In this branch this heating coil operates at PLR = 0.991653 versus PLR = 0.518 in develop.

image

**Comment by Myoldmopar** [comment]
Thanks for the walk through of this today.  The change makes sense, and the results look great.  Getting a PLR of close to 1 makes great sense, and indicates that the sizing algorithms are properly sizing now.  Merging this one.

**Pull request merged**